src/content/app/genome-browser/hooks/useGenomeBrowserAnalytics.ts

```diff
@@ -0,0 +1,307 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useSelector } from 'react-redux';
+
+import analyticsTracking from 'src/services/analytics-service';
+
+import useGenomeBrowserIds from './useGenomeBrowserIds';
+
+import { getSpeciesAnalyticsName } from 'src/content/app/species-selector/speciesSelectorHelper';
+import { parseFocusObjectId } from 'src/shared/helpers/focusObjectHelpers';
+
+import { getCommittedSpeciesById } from 'src/content/app/species-selector/state/speciesSelectorSelectors';
+
+import { AppName } from 'src/global/globalConfig';
+import { Status } from 'src/shared/types/status';
+
+import type { AnalyticsOptions } from 'src/analyticsHelper';
+import type { TrackSet } from '../components/track-panel/trackPanelConfig';
+import type { RootState } from 'src/store';
+
+const useGenomeBrowserAnalytics = () => {
+  const { activeFocusObjectId, activeGenomeId } = useGenomeBrowserIds();
+
+  const committedSpecies = useSelector((state: RootState) =>
+    getCommittedSpeciesById(state, activeGenomeId)
+  );
+  const speciesNameForAnalytics = committedSpecies
+    ? getSpeciesAnalyticsName(committedSpecies)
+    : '';
+
+  const { type = '', objectId = '' } = activeFocusObjectId
+    ? parseFocusObjectId(activeFocusObjectId)
+    : {};
+
+  const feature = activeFocusObjectId ? `${type} - ${objectId}` : '';
+
+  const sendTrackEvent = (options: AnalyticsOptions) => {
+    analyticsTracking.trackEvent({
+      species: speciesNameForAnalytics,
+      feature,
```

Comment:
We talked about this during code review; but I didn't quite understand the answer. Let's say we had an extra flag in the `options` that would dictate whether we add `feature` to the analytics payload or not (so that only the events that needed the feature would report it) — would GA dashboard still show the `feature` dimension for all events, even for those that didn't send it to analytics?

Reply:
Yes, that's my understanding. These custom dimensions will just be empty for the events we don't set them.